### PR TITLE
Add Mono to the repository

### PR DESCRIPTION
**Pull request by akoeplinger**
This adds the Mono runtime sources including the full history and some initial build system integration 🎉 

We're planning to wire up the CI on Monday and then we'll enable Mono by default as part of the build.

/cc @dotnet/runtime-infrastructure @jkotas @danmosemsft @jaredpar

### CODE REVIEW

**Comment by directhex** [comment]
For the love of $DEITY, remember to merge, not squash!

**Comment by xamarindevelopervietnam** [comment]
This’s great step!

**Review by lewurm — APPROVED** [review]

**Review by jkotas — APPROVED** [review]

**Comment by akoeplinger** [comment]
I pressed the merge button and https://github.com/dotnet/runtime/commit/1cb1285408579d53a31bd7659faba53eae3d4262 was committed to master :+1:

Looks like this PR was a little to heavy for GitHub to automatically close so I'm manually closing it :)

File: eng/testing/tests.targets [hunk 1]
```diff
@@ -72,6 +74,19 @@
...
+
+    <!-- Generate Mono .rsp file -->
+    <ItemGroup>
+      <MonoRspFiles Include="$(MonoProjectRoot)netcore\CoreFX.issues.rsp" />
+      <MonoRspFiles Include="$(MonoProjectRoot)netcore\CoreFX.issues_mac.rsp" Condition="'$(TargetOS)' == 'OSX'" />
+      <MonoRspFiles Include="$(MonoProjectRoot)netcore\CoreFX.issues_linux.rsp" Condition="'$(TargetOS)' == 'Linux'" />
+      <MonoRspFiles Include="$(MonoProjectRoot)netcore\CoreFX.issues_windows.rsp" Condition="'$(TargetOS)' == 'Windows_NT'" />
+    </ItemGroup>
+    <ItemGroup>
+      <MonoRspFileContent Include="$([System.IO.File]::ReadAllText(%(MonoRspFiles.Identity)))" />
```
Comment (ViktorHofer) [inline eng/testing/tests.targets]: Should this be conditioned to only copy the rsp file when targeting Mono?

Comment (akoeplinger) [reply eng/testing/tests.targets]: We can do that, sure.

Comment (marek-safar) [reply eng/testing/tests.targets]: I think it'd be better if the files were located under src/libraries

Comment (ViktorHofer) [reply eng/testing/tests.targets]: We agreed on using Attributes instead. Is it feasible to remove these rsp files all together now?

Comment (marek-safar) [reply eng/testing/tests.targets]: Not now, we still have to annotate many tests (100's+) with different attributes

**Pull request closed without merging**